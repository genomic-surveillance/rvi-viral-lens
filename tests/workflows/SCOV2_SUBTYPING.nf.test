// Copyright (C) 2023 Genome Surveillance Unit/Genome Research Ltd.

nextflow_workflow {

    name "Test Workflow SCOV2_SUBTYPING"
    script "workflows/SCOV2_SUBTYPING.nf"
    workflow "SCOV2_SUBTYPING"

    test("Should run without failures") {

        when {
            params {
                meta = [
                    id:"test.2222222",
                    sample_id:"test",
                    taxid:2222222
                ]
                test_loc="${projectDir}/tests/test_data"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = tuple(
                    params.meta,
                    "${params.test_loc}/test_consensuses/test2_sarscov2.fa")
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
