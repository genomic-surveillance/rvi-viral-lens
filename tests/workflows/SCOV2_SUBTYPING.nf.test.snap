{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.2222222",
                            "taxid": 2222222,
                            "sample_id": "test",
                            "virus_subtype": "EG.5.1.1"
                        }
                    ]
                ],
                "scov2_subtype_out_ch": [
                    [
                        {
                            "id": "test.2222222",
                            "taxid": 2222222,
                            "sample_id": "test",
                            "virus_subtype": "EG.5.1.1"
                        }
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "22.04.5"
        },
        "timestamp": "2024-03-04T16:54:58.252456"
    }
}